File: BMICalculator/app/src/main/java/com/example/bmicalculator/BmrCalculator.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;

public class BmrCalculator extends AppCompatActivity {

    EditText weight, height;
    String bmiResult = HomeActivity.getBMIresult();
    String dish,dishprop;
    EditText a;
    float bmr = 0;
    TextView result,resultDish;
    String calculation;
    private Button bmrButton;
    RadioGroup radioGender;
    String gender;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmr_calculator);
        a = findViewById(R.id.age);
        result = findViewById(R.id.result);
        resultDish = findViewById(R.id.resultDish);



    }
    public void onRadioButtonClicked(View v) {
        switch (v.getId()) {
            case R.id.male:
                gender = "m";
                break;
            case R.id.female:
                gender = "f";
                break;
        }
    }
    public void calculateBMR(View view) {
        float weight = Float.parseFloat(HomeActivity.getWeight().getText().toString());
        float height = Float.parseFloat(HomeActivity.getHeight().getText().toString());
        int age = Integer.parseInt(a.getText().toString());

        if (gender == "m") {
            bmr = 66 + (13.7f * weight) + (5 * height) - (6.8f * age);
        } else if (gender == "f") {
            bmr = 655 + (9.6f * weight) + (1.8f * height) - (4.7f * age);
        } else {
            calculation = "Check your gender";
            bmr = 2;
        }

        calculation = "BMR = " + String.format("%,.2f", bmr);
        if (weight <= 0.0f || height <= 0.0f) {
            calculation = "Calculate BMI first";
        }

        propositeRecipe();
        result.setText(calculation);
    }

    public void propositeRecipe() {
        switch (bmiResult) {
            case "Underweight":
                dish = "Kebab";
                break;
            case "Healthy Weight":
                dish = "Eat what you want";
                break;
            case "Overweight":
                dish = "Chicken with rice";
                break;
            case "Obese":
                dish = "Mixed Salad";
                break;
            default:
                dish = "Error";
        }

        dishprop = "Dinner proposition:"+dish;
      resultDish.setText(dishprop);
    }


    public void gotoQuiz(View view) {
        Intent QuizIntent = new Intent(BmrCalculator.this, Quiz.class);
        startActivity(QuizIntent);

    }
}

File: BMIKotlin/BMICalculator/app/src/main/java/com/example/bmicalculator/BmrCalculator.kt
package com.example.bmicalculator

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.bmicalculator.R.id.bmrcalc
import com.example.bmicalculator.R.id.calculateBMR


class BmrCalculator : AppCompatActivity() {

    var weight: EditText? = null
    var height: EditText? = null
    var bmiResult: String? = intent.getStringExtra("BmiResult")
    var dish: String? = null
    var dishprop: String? = null
    var a: EditText? = null
    var bmr = 0f
    var result: TextView? = null
    var resultDish: TextView? = null
    var calculation: String? = null
     var bmrButton: Button? = null
    var radioGender: RadioGroup? = null
    var gender: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmr_calculator)
        a = findViewById(R.id.age)
        result = findViewById(R.id.result)
        resultDish = findViewById(R.id.resultDish)
         val bmrcalcButton = findViewById<Button>(R.id.calculateBMR)

        bmrcalcButton.setOnClickListener(){
            val weightt: Float = weight.toString().toFloat()
            val heightt: Float = height.toString().toFloat()
            val age: Int = a!!.text.toString().toInt()
            if (gender === "m") {
                bmr = 66 + 13.7f * weightt + 5 * heightt - 6.8f * age
            } else if (gender === "f") {
                bmr = 655 + 9.6f * weightt + 1.8f * heightt - 4.7f * age
            } else {
                calculation = "Check your gender"
                bmr = 2f
            }
            calculation = "BMR = " + String.format("%,.2f", bmr)
            if (weightt <= 0.0f || heightt <= 0.0f) {
                calculation = "Calculate BMI first"
            }
            propositeRecipe()
            result!!.text = calculation

        }
    }

    fun onRadioButtonClicked(v: View) {
        when (v.id) {
            R.id.male -> gender = "m"
            R.id.female -> gender = "f"
        }
    }




    fun propositeRecipe() {
        dish = when (bmiResult) {
            "Underweight" -> "Kebab"
            "Healthy Weight" -> "Eat what you want"
            "Overweight" -> "Chicken with rice"
            "Obese" -> "Mixed Salad"
            else -> "Error"
        }
        dishprop = "Dinner proposition:$dish"
        resultDish!!.text = dishprop
    }

    fun gotoQuiz() {
        val QuizIntent = Intent(this@BmrCalculator, Quiz::class.java)
        startActivity(QuizIntent)
    }

}